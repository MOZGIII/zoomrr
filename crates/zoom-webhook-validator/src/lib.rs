use hmac::Mac;

type HmacSha256 = hmac::Hmac<sha2::Sha256>;

#[derive(Debug)]
pub struct ZoomSignedWebhookValidator {
    mac: HmacSha256,
}

impl ZoomSignedWebhookValidator {
    pub fn new(secret_token: &str) -> Self {
        let mac = HmacSha256::new_from_slice(secret_token.as_bytes()).unwrap();
        Self { mac }
    }
}

impl ZoomSignedWebhookValidator {
    pub fn encrypt(&self, token: &str) -> String {
        let mut mac = self.mac.clone();
        mac.update(token.as_bytes());
        let result = mac.finalize();
        let code_bytes = result.into_bytes();
        format!("{:x}", code_bytes)
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn zoom_sample() {
        let validator = ZoomSignedWebhookValidator::new("test");
        let token = "qgg8vlvZRS6UYooatFL8Aw";
        let value = validator.encrypt(token);
        insta::assert_snapshot!(value);
    }
}
