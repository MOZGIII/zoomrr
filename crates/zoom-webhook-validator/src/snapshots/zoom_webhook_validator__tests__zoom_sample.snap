---
source: crates/zoom-webhook-validator/src/lib.rs
expression: value
---
ae859bbc963b7d8fbc9cffbc482b0ea7a4486f1fc23e29217744236dd6e4d6d8
